<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="launchprefix" default="gdbserver localhost:10000"/>

  <!-- topic parameters -->
  <arg name="topic_pose_in"             default="/mus/ground_truth_pose_imu" />
  <arg name="topic_uwb_in"              default="/mus/uwb" />
  <arg name="topic_anchor_out"          default="~anchors" />
  <arg name="topic_wp_out"              default="~waypoints" />

  <!-- calibration -->
  <arg name="calib_p_ItoU"              default="[0.0, 0.0, 0.0]" />

  <!-- node parameters -->
  <arg name="num_anchors"               default="4" />
  <arg name="buffer_size_in_s"          default="50.0" />
  <arg name="max_cond_number"           default="300.0" />
  <arg name="init_check_duration_in_s"  default="5.0" />
  <arg name="t_pose_diff_s"             default="0.0" />

  <arg name="meas_baseline_m"           default="0.3" />
  <arg name="meas_baseline_idx"         default="50" />
  <arg name="reg_lambda"                default="100" />
  <arg name="continous_init"            default="False" />

  <!-- waypoint generation -->
  <arg name="wp_max_dist_in_m"          default="1.0" />
  <arg name="wp_height_in_m"            default="2.0" />


  <node name="uwb_init_cpp" pkg="uwb_init_cpp" type="uwb_init_cpp" args="" output="screen" respawn="false" launch-prefix="$(arg launchprefix)">
    <!-- topics remaps -->
    <param name="topic_sub_pose"      type="str"  value="$(arg topic_pose_in)" />
    <param name="topic_sub_uwb"       type="str"  value="$(arg topic_uwb_in)" />
    <param name="topic_pub_anchors"   type="str"  value="$(arg topic_anchor_out)" />
    <param name="topic_pub_waypoints" type="str"  value="$(arg topic_wp_out)" />

    <!-- node parameters -->
    <param name="init_check_duration"   type="double" value="$(arg init_check_duration_in_s)" />
    <param name="n_anchors"             type="int"    value="$(arg num_anchors)" />
    <param name="buffer_size_s"         type="double" value="$(arg buffer_size_in_s)" />
    <param name="max_cond_num"          type="double" value="$(arg max_cond_number)" />
    <rosparam param="p_ItoU"            subst_value="True">$(arg calib_p_ItoU)</rosparam>
    <param name="t_pose_diff_s"         type="double" value="$(arg t_pose_diff_s)" />

    <param name="meas_baseline_m"       type="double" value="$(arg meas_baseline_m)" />
    <param name="meas_baseline_idx"     type="int" value="$(arg meas_baseline_idx)" />
    <param name="reg_lambda"            type="double" value="$(arg reg_lambda)" />
    <param name="do_continous_init"     type="bool" value="$(arg continous_init)" />

    <!-- waypoint generation -->
    <param name="waypoint_max_dist_m"   type="double"   value="$(arg wp_max_dist_in_m)" />
    <param name="waypoint_height_m"     type="double"   value="$(arg wp_height_in_m)" />
  </node>
</launch>
